revocation/ocsp/ocsp.go

```diff
@@ -167,17 +167,25 @@ func extensionsToMap(extensions []pkix.Extension) map[string][]byte {
 func executeOCSPCheck(cert, issuer *x509.Certificate, server string, opts Options) (*ocsp.Response, error) {
 	// TODO: Look into other alternatives for specifying the Hash
 	// https://github.com/notaryproject/notation-core-go/issues/139
-	ocspRequest, err := ocsp.CreateRequest(cert, issuer, &ocsp.RequestOptions{Hash: crypto.SHA256})
+	// The following do not support SHA256 hashes:
+	//  - Microsoft
+	//  - Entrust
+	//  - Let's Encrypt
+	//  - Digicert (sometimes)
+	// As this represents a large percentage of public CAs, we are using the
+	// hashing algorithm SHA1, which has been confirmed to be supported by all
+	// that were tested.
+	ocspRequest, err := ocsp.CreateRequest(cert, issuer, &ocsp.RequestOptions{Hash: crypto.SHA1})
 	if err != nil {
 		return nil, GenericError{Err: err}
 	}
 
 	var resp *http.Response
-	if base64.URLEncoding.EncodedLen(len(ocspRequest)) >= 255 {
+	if base64.StdEncoding.EncodedLen(len(ocspRequest)) >= 255 {
```

Comment:
The RFC2560 states

> To enable HTTP caching, small requests (that
   after encoding are less than 255 bytes), MAY be submitted using GET.

But I am not sure which encoding it refers to. If it refers to `url-encoding of base-64 encoding of the DER encoding of
the OCSPRequest`, we need an efficient way to calculate the encoded length at least with memory complexity of `O(1)`.

Reply:
My guess is that it is before url-encoding. However, url-encoding would be longer, so I'd say let's use that to be safe. One way that would have memory O(1) but time O(n) would be counting the number of characters that would be escaped and adding 2x that number to the length (since each replacement is with a %XX)

Reply:
Actually, this would require us to store the encoded URL. I think it is best to just check the base64 encoded length first, then only create the path escaped one if that is okay and confirm the length of that. (This would mean it couldn't ever exceed a max size of 255*3 because it would need to be base64 encoded to < 255 and each replacement would be at most 3 characters replacing 1, so technically O(1) since there is an upper bound)